service.go

```diff
@@ -0,0 +1,33 @@
+package hsm
+
+import "go.uber.org/zap"
+
+type Service interface {
```

Comment:
I'm not sure if this is needed to be part of this package, as none of the pieces of this package use it. This may be used by an implementation of the hsm package, but the Service itself seems like a separate component. This style of Get and Set on interfaces is pretty non-idiomatic as you lose all sorts of type safety, so I'd avoid including it if it isn't used.

Reply:
Agreed.